java/component/keyple-plugin/remote-se/core/src/main/java/org/eclipse/keyple/plugin/remotese/core/impl/KeypleClientAsyncNodeImpl.java

```diff
@@ -0,0 +1,510 @@
+/********************************************************************************
+ * Copyright (c) 2020 Calypso Networks Association https://www.calypsonet-asso.org/
+ *
+ * See the NOTICE file(s) distributed with this work for additional information regarding copyright
+ * ownership.
+ *
+ * This program and the accompanying materials are made available under the terms of the Eclipse
+ * Public License 2.0 which is available at http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ ********************************************************************************/
+package org.eclipse.keyple.plugin.remotese.core.impl;
+
+import java.util.*;
+import org.eclipse.keyple.core.util.Assert;
+import org.eclipse.keyple.plugin.remotese.core.KeypleClientAsync;
+import org.eclipse.keyple.plugin.remotese.core.KeypleClientAsyncNode;
+import org.eclipse.keyple.plugin.remotese.core.KeypleMessageDto;
+import org.eclipse.keyple.plugin.remotese.core.exception.KeypleTimeoutException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Keyple Client Async Node implementation.
+ * <p>
+ * This is an internal class an must not be used by the user.
+ *
+ * @since 1.0
+ */
+public final class KeypleClientAsyncNodeImpl extends AbstractKeypleNode
+        implements KeypleClientAsyncNode {
+
+    private static final Logger logger = LoggerFactory.getLogger(KeypleClientAsyncNodeImpl.class);
+
+    private final KeypleClientAsync endpoint;
+    private final Map<String, SessionManager> sessionManagers;
+
+    // Timeout used during awaiting (in milliseconds)
+    private final int timeout;
+
+    /**
+     * (package-private)<br>
+     * Constructor.
+     *
+     * @param handler The associated handler (must be not null).
+     * @param endpoint The user client async endpoint (must be not null).
+     * @param timeoutInSecond The default timeout (in seconds) to use.
+     */
+    KeypleClientAsyncNodeImpl(AbstractKeypleMessageHandler handler, KeypleClientAsync endpoint,
+            int timeoutInSecond) {
+        super(handler);
+        this.endpoint = endpoint;
+        this.timeout = timeoutInSecond * 1000;
+        this.sessionManagers = new HashMap<String, SessionManager>();
```

Comment:
ConcurrentHashMap ?

Reply:
No need because there is no itaration on entries and the only actions are put and delete.